app/org/sagebionetworks/bridge/validators/UploadSchemaValidator.java

```diff
@@ -89,6 +94,17 @@ public void validate(Object target, Errors errors) {
             } else {
                 UploadFieldDefinitionListValidator.INSTANCE.validate(fieldDefList, errors, "fieldDefinitions");
             }
+
+            // fieldDef max size
+            UploadFieldSize fieldSize = UploadUtil.calculateFieldSize(fieldDefList);
+            if (fieldSize.getNumBytes() > MAX_BYTES) {
+                errors.rejectValue("fieldDefinitions",
+                        "cannot be greater than 50000 bytes combined");
+            }
+            if (fieldSize.getNumColumns() > MAX_COLUMNS) {
+                errors.rejectValue("fieldDefinitions",
+                        "cannot be greater than 100 columns combined");
```

Comment:
152 cols Synapse limitation - (100 schema cols + 20 shared metadata cols) = 32 columns that are left for what?

Reply:
See https://sagebionetworks.jira.com/wiki/spaces/BRIDGE/pages/763166825/Bridge+Table+Width+Limits for Bridge limits. This will be moved to external-facing documentation once this has been pushed out to production.